api/v1/verticadb_webhook.go

```diff
@@ -2059,6 +2111,27 @@ func (v *VerticaDB) checkImmutableStsName(oldObj *VerticaDB, allErrs field.Error
 	return allErrs
 }
 
+// checkImmutableClientProxy will validate the proxy spec fields in vdb
+func (v *VerticaDB) checkImmutableClientProxy(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	// annotation vertica.com/use-client-proxy cannot change after creation
+	if v.Annotations[vmeta.UseVProxyAnnotation] != oldObj.Annotations[vmeta.UseVProxyAnnotation] {
+		prefix := field.NewPath("metadata").Child("annotations")
+		errMsg := fmt.Sprintf("annotation %s cannot change after creation", vmeta.UseVProxyAnnotation)
+		err := field.Invalid(prefix.Key(vmeta.UseVProxyAnnotation),
+			v.Annotations[vmeta.VClusterOpsAnnotation],
+			errMsg)
+		allErrs = append(allErrs, err)
+	}
+	// proxy.image cannot change after creation
+	if v.Spec.Proxy != nil && oldObj.Spec.Proxy != nil && v.Spec.Proxy.Image != oldObj.Spec.Proxy.Image {
+		err := field.Invalid(field.NewPath("spec").Child("proxy").Child("image"),
+			v.Spec.Proxy.Image,
+			"proxy.image cannot change after creation")
```

Comment:
`proxy.image cannot change after creation, otherwise, as doing so will disrupt current user connections`

Reply:
Updated in https://github.com/vertica/vertica-kubernetes/pull/1000/commits/42ee9713632436e3a9b76048fe6e12035d6bc83d